modules/proposals/engine/src/lib.rs

```diff
@@ -133,20 +144,21 @@ decl_module! {
         fn deposit_event() = default;
 
         /// Vote extrinsic. Conditions:  origin must allow votes.
-        pub fn vote(origin, proposal_id: u32, vote: VoteKind)  {
-            let voter_id = T::VoteOrigin::ensure_origin(origin)?;
+        pub fn vote(origin, proposal_id: T::ProposalId, vote: VoteKind)  {
+            let account_id = T::VoteOrigin::ensure_origin(origin)?;
+            let voter_id = T::VoterId::from(account_id);
 
             ensure!(<Proposals<T>>::exists(proposal_id), errors::MSG_PROPOSAL_NOT_FOUND);
-            let proposal = Self::proposals(proposal_id);
+            let mut proposal = Self::proposals(proposal_id);
 
             let not_expired = !proposal.is_voting_period_expired(Self::current_block());
```

Comment:
Why do we check this expiration separately from checking the status of the proposal? the status representation covers this case? It looks like a bug. The only case I can think of which sort of could make sense is if we are worried about the proposal expiring in this very same block, but the expiration handler being executed later in `on_finalized`. If this is the case you are worried about, then I suggest we ignore it, because its a purely semantic issue what it means to "expire in block x", it could mean at the end of the block, in which case this double-check goes away.

Reply:
This is a redundant check. Will remove.